lib/tasks/links_migration/link_migrator.rb

```diff
@@ -0,0 +1,83 @@
+module Tasks
+  module LinksMigration
+    class LinkMigrator
+      class DodgyBasePathError < StandardError; end
+
+      def populate_topic_links
+        relevant_subscriber_lists.each do |list|
+          content_item = Services.content_store.content_item(base_path_from(list))
+
+          if content_item.blank?
+            raise DodgyBasePathError, <<-ERROR.strip_heredoc
+              No content item found for #{base_path_from(list)},
+              run `bundle exec rake links_migration:report_non_matching`
+              and fix these cases before continuing migration.
+            ERROR
+          end
+
+          if content_item.content_id.blank?
+            raise DodgyBasePathError, <<-ERROR.strip_heredoc
+              No content ID found for #{base_path_from(list)},
+              run `bundle exec rake links_migration:report_non_matching`
+              and fix these cases before continuing migration.
+            ERROR
+          end
+
```

Comment:
I was wrong. It's doing it here: https://github.com/alphagov/content-store/blob/master/app/models/content_item.rb#L216